pkg/frontend/mysql_protocol.go

```diff
@@ -984,27 +990,32 @@ func (mp *MysqlProtocolImpl) setSequenceID(value uint8) {
 	mp.sequenceId = value
 }
 
-func (mp *MysqlProtocolImpl) handleHandshake(payload []byte) error {
+func (mp *MysqlProtocolImpl) handleHandshake(payload []byte) (bool, error) {
 	if len(payload) < 2 {
-		return fmt.Errorf("received a broken response packet")
+		return false, fmt.Errorf("received a broken response packet")
 	}
 
 	var authResponse []byte
 	if capabilities, _, ok := mp.io.ReadUint16(payload, 0); !ok {
-		return fmt.Errorf("read capabilities from response packet failed")
+		return false, fmt.Errorf("read capabilities from response packet failed")
 	} else if uint32(capabilities)&CLIENT_PROTOCOL_41 != 0 {
 		var resp41 response41
 		var ok bool
 		var err error
 		if ok, resp41, err = mp.analyseHandshakeResponse41(payload); !ok {
-			return err
+			return false, err
+		}
+
+		// client ask server to upgradeTls
+		if len(resp41.username) == 0 && resp41.capabilities&CLIENT_SSL != 0 {
```

Comment:
Why does it need to check the user name? The reference of the mysql does not mention the username.